compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -9996,23 +9996,25 @@ private STNode parseClientDeclOrVarDeclStatement(STNode metadata, List<STNode> q
     }
 
     private STNode parseClientDeclOrVarDecl(STNode metadata, STNode publicQualifier, List<STNode> qualifiers,
-                                            boolean isModuleVar) {
+                                            boolean moduleDecl) {
         STToken nextToken = peek();
         switch (nextToken.kind) {
             case STRING_LITERAL_TOKEN:
                 reportInvalidQualifiersOnClientDecl(publicQualifier, qualifiers);
-                return parseClientDeclaration(getAnnotations(metadata), qualifiers.get(qualifiers.size() - 1),
-                                              isModuleVar);
+                return parseClientDeclaration(moduleDecl && metadata != null ?
```

Comment:
```suggestion
                return parseClientDeclaration(metadata != null && metadata.kind == SyntaxKind.METADATA ?
```
Let's get annotations to a separate variable. We need to invalidate any documentation inside the metadata node. Hence create a method like `invalidateDocStringAndGetAnnotations()`

Reply:
Thanks, updated in def9f45 to introduce `invalidateDocumentation`. Didn't move the annotation logic there though since it's only applicable for module-level declarations.